part-2-server/server.js

```diff
@@ -0,0 +1,43 @@
+var http = require("http");
+var fs = require("fs");
+
+var server = http.createServer(function(req, res) {
+    if (req.url.match(/\/poem-/)) {
+        var target = req.url.split("");
+        target.shift();
+        target = target.join("");
+        target = target + ".txt"
+        fs.readFile(target, function(err, data) {
+            res.write(data.toString());
+            res.end();
+        })
+    }
+
```

Comment:
URL requests that start with 'poem' will hit the first if statement, then will hit the switch statement and eventually hit the default case. This will cause your server to crash because within the first if statement it will try to read the file asyncronously, continue on to the switch statement, then set a response.end before the original file is read